File: Arrays & Hashing/217. Contains Duplicate/Solution.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # N : len(nums)
        # Time Complexity: O(N)
        # Space Comlpexity: O(N)
        numSet = set()
        for num in nums:
            if num in numSet:
                return True
            numSet.add(num)
        return False

File: Arrays & Hashing/217. Contains Duplicate/Solution.java
import java.util.*;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        // N : len(nums)
        // Time Complexity: O(N)
        // Space Comlpexity: O(N)
        HashSet numSet = new HashSet<>();
        for (int num: nums) {
            if (numSet.contains(num)) return true;
            numSet.add(num);
        }
        return false;
    }
}